pkg/api/smartcontracts/update_smartcontract_handler.go

```diff
@@ -0,0 +1,73 @@
+package smartcontracts
+
+import (
+	"encoding/json"
+
+	"github.com/darchlabs/synchronizer-v2/pkg/smartcontract"
+	"github.com/go-playground/validator"
+	"github.com/gofiber/fiber/v2"
+)
+
+type updateSmartContractResponse struct {
+	Data  *smartcontract.SmartContract `json:"data"`
+	Error string                       `json:"error,omitempty"`
+}
+
+func updateSmartContractHandler(ctx Context) func(c *fiber.Ctx) error {
+	return func(c *fiber.Ctx) error {
+		c.Accepts("application/json")
+
+		address := c.Params("address")
+		if address == "" {
+			return c.Status(fiber.StatusOK).JSON(updateSmartContractResponse{
+				Error: "address cannot be nil",
+			})
+		}
+
+		// prepate body request struct
+		body := struct {
+			SmartContract *smartcontract.SmartContractUpdate `json:"smartcontract"`
+		}{}
+
+		// parse body to smartcontract struct
+		err := json.Unmarshal(c.Body(), &body)
```

Comment:
fiber has builtin functions for make this more reliable and avoid own-handling errors